scripts/dev/tidy.php

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env php
+<?php
+
+/*
+ * +----------------------------------------------------------------------+
+ * | PHP Version 7                                                        |
+ * +----------------------------------------------------------------------+
+ * | Copyright (c) 1997-2018 The PHP Group                                |
+ * +----------------------------------------------------------------------+
+ * | This source file is subject to version 3.01 of the PHP license,      |
+ * | that is bundled with this package in the file LICENSE, and is        |
+ * | available through the world-wide-web at the following url:           |
+ * | https://php.net/license/3_01.txt                                     |
+ * | If you did not receive a copy of the PHP license and are unable to   |
+ * | obtain it through the world-wide-web, please send a note to          |
+ * | license@php.net so we can mail you a copy immediately.               |
+ * +----------------------------------------------------------------------+
+ *
+ * This is a tidy script that checks and fixes trailing whitespace and final
+ * new lines in a given path.
+ *
+ * For PHP source code repository several exclusions are defined such as
+ * binary files, and certain test files.
+ *
+ * See usage function below for information how to use it.
+ */
+
+function usage(): string
+{
+    return <<<EOT
+PHP tidy script
+
+This script checks and fixes trailing whitespace and final new lines in a
+given path.
+
+USAGE:
+  php tidy.php [OPTION...] input-path
+
+OPTIONS:
+  -h, --help          Display this help
+  -v, --version       Display script version
+      --no-whitespace Disable cleaning whitespace
+      --no-eol        Disable cleaning final end of lines
+  -t, --trim          Write trimmed files in place instead of only reporting them
+
+EOT;
+}
+
+// List of excluded directories and exact filenames
+$excludePaths = [
+    '.git',
+
+    // Files that are generated from other files
+    'Zend/zend_language_scanner.c',
+    'Zend/zend_ini_scanner.c',
+    'Zend/zend_vm_execute.h',
+    'Zend/zend_vm_opcodes.h',
+    'Zend/zend_vm_opcodes.c',
+    'sapi/cli/mime_type_map.h',
+    'sapi/phpdbg/phpdbg_lexer.c',
+    'ext/json/json_scanner.c',
+    'ext/pdo/pdo_sql_parser.c',
+    'ext/standard/url_scanner_ex.c',
+    'ext/standard/var_unserializer.c',
+    'ext/phar/phar_path_check.c',
+    'ext/fileinfo/data_file.c',
+    'ext/standard/credits_ext.h',
+    'ext/standard/credits_sapi.h',
+    'ext/mbstring/unicode_data.h',
+    'win32/cp_enc_map.c',
+    'ext/standard/html_tables.h',
+
+    // Test folders
+    'Zend/tests',
+    'tests',
+    'sapi/tests',
+
+    // Bundled libraries maintained
```

Comment:
Missing a word at the end :)

Reply:
Updated with `Bundled libraries maintained in the upstream repositories`.